### ISSUE TITLE
[Question] PPO1 and PPO2 hyperparameters on custom environments

### ISSUE THREAD

**Issue by alitk** [hill-a/stable-baselines#442]
Hi
I created a custom environment and it works perfectly with OpenAI Spinning Up PPO [here](https://spinningup.openai.com/en/latest/algorithms/ppo.html). However, it does not even start to converge using both PPO1 and PPO2? what do you think can be the problem. What would be the first hyperparameter that you would check?

**Comment by araffin** [comment]
Hello,
Did you try using the same hyperparameters as spinning up? (and don't forget the number of actors ;))

**Comment by Tomas0413** [comment]
@alitk have you made progress on this issue?

I have also faced a similar issue with a custom environment. I ended up comparing the default parameter values of Spinning Up PPO and Stable Baselines PPO2.

And I ended up increasing the value for n_steps and noptepochs.
This worked in my case pretty well, but I'm going to run a few more experiments and will attempt to understand the PPO algorithm in general.

Before:
model = PPO2(MlpPolicy, env, verbose=1, tensorboard_log='/path/to/tensorboard/logs')

After:
model = PPO2(MlpPolicy, env, verbose=1, tensorboard_log='/path/to/tensorboard/logs', n_steps=20000, noptepochs=50)

**Comment by Miffyli** [comment]
I would to like highlight that other parameters may also have a significant effect: Sometimes other network initializations work much faster, sometimes even very high learning rate is ok and works out of the box, advantage computation ("normal" advantage vs. GAE, used by PPO), so on and so on. Personally I have not had luck replicating results exactly when trying to implement same code in another framework, for example.
